src/logic/voice/voice.ts

```diff
@@ -153,12 +242,167 @@ export function createVoiceComponent(
+    }
+  }
+
+  /**
+   * Generates credentials for a community voice chat room for a member.
+   */
+  async function getCommunityVoiceChatCredentialsForMember(
+    communityId: string,
+    userAddress: string
+  ): Promise<{ connectionUrl: string }> {
+    const roomName = getCommunityVoiceChatRoomName(communityId)
+
+    const roomKey = await livekit.generateCredentials(
+      userAddress,
+      roomName,
+      {
+        cast: [],
+        canPublish: false, // can't publish until they are promoted
+        canSubscribe: true,
+        canUpdateOwnMetadata: false
+      },
+      false,
+      {
+        role: CommunityRole.Member,
+        community_id: communityId
+      }
+    )
+
+    // Add member to the community room
+    await voiceDB.joinUserToCommunityRoom(userAddress, roomName, false)
+
+    return {
+      connectionUrl: `livekit:${roomKey.url}?access_token=${roomKey.token}`
+    }
+  }
+
+  /**
+   * Deletes expired community voice chats.
+   * THIS IS THE KEY FUNCTION - runs every minute to check for rooms with no moderators for 5+ minutes
+   */
+  async function expireCommunityVoiceChats(): Promise<void> {
+    logger.debug('Running community voice chat expiration job')
+    const expiredRoomNames = await voiceDB.deleteExpiredCommunityVoiceChats()
+
+    logger.debug(`Found ${expiredRoomNames.length} expired community voice chat rooms`)
+
+    // Delete the expired rooms from LiveKit.
+    for (const roomName of expiredRoomNames) {
+      const communityId = getCommunityIdFromRoomName(roomName)
+      logger.info(`Expiring community voice chat room: ${roomName} (community: ${communityId})`)
+
+      analytics.fireEvent(AnalyticsEvent.EXPIRE_CALL, {
+        call_id: communityId
+      })
+
+      await livekit.deleteRoom(roomName)
+    }
+  }
+
+  /**
+   * Gets the status of a community voice chat.
+   * Uses stored database information about users joining/leaving to determine if the room is active.
+   * Includes grace period for moderators who suffered connection interruptions.
+   */
+  async function getCommunityVoiceChatStatus(communityId: string): Promise<{
+    active: boolean
+    participantCount: number
+    moderatorCount: number
+  }> {
+    const roomName = getCommunityVoiceChatRoomName(communityId)
+
+    logger.debug(`Getting status for community voice chat: ${roomName}`)
+
+    try {
+      const usersInRoom = await voiceDB.getCommunityUsersInRoom(roomName)
+
+      if (usersInRoom.length === 0) {
+        logger.debug(`Community voice chat room ${roomName} has no users in database`)
+        return {
+          active: false,
+          participantCount: 0,
+          moderatorCount: 0
+        }
+      }
+
+      // Count active participants (connected or recently interrupted)
+      const activeParticipants = usersInRoom.filter(
+        (user) =>
+          user.status === VoiceChatUserStatus.Connected || user.status === VoiceChatUserStatus.ConnectionInterrupted
```

Comment:
Are we considering active participants the users that have their connection being interrupted longer than the expected TTL? Maybe we can create another function that re-uses part of the logic in the `shouldDestroyCommunityRoom` function.